engine/runtime/src/main/scala/org/enso/interpreter/runtime/util/DiagnosticFormatter.scala

```diff
@@ -200,31 +323,81 @@ class DiagnosticFormatter(
     fansi.Str("^" + ("~" * sectionLen)).overlay(textAttrs)
   }
 
-  private def fileLocationFromSectionOption(
-    loc: Option[IdentifiedLocation],
-    source: Source
-  ): String = {
-    val srcLocation = loc match {
-      case Some(identifiedLoc)
-          if isLocationInSourceBounds(identifiedLoc, source) =>
-        val section =
-          source.createSection(identifiedLoc.start, identifiedLoc.length)
-        val locStr =
-          "" + section.getStartLine + ":" +
-          section.getStartColumn + "-" +
-          section.getEndLine + ":" +
-          section.getEndColumn
-        "[" + locStr + "]"
-      case _ => ""
-    }
-
-    source.getPath + ":" + srcLocation
-  }
-
-  private def isLocationInSourceBounds(
-    loc: IdentifiedLocation,
-    source: Source
-  ): Boolean = {
-    loc.end() <= source.getLength
+  private def includeGithubAnnotation: Boolean =
+    sys.env.get("GITHUB_ACTIONS").contains("true")
```

Comment:
- can we somehow separate this "GitHub Actions" code from regular Enso CLI code?
- @Akirathan suggested to put it into completely different place, but I am afraid of that being too complex
- however a simple change would be to introduce `GitHubDiagnosticsFormatter extends DiagnosticsFormatter`
- and put this "GitHub Actions" code there.
- such a change would be a mere refactoring
- and it would keep the `DiagnosticsFormatter` almost simple

Reply:
Re. [Radek's reply](https://github.com/enso-org/enso/pull/12901#issuecomment-28270527230):

> reporting the diagnostics in some common format (JSON?) and then parsing them to generate the annotations. But I think this is just overkill.

Yes, that is an overkill. Simple refactoring of the code will do.

Reply:
Yes that is a great suggestion, after @Akirathan I also considered extracting the GH-specific parts into a separate class, so I will try to do something like that.

I agree that this change to this file grew out of proportion and it is violating SRP now.

Reply:
Refactored in 48327a6b4132fff7c130c410e648a31d80b13276 (with some small followup changes)